base/twiceprecision.jl

```diff
@@ -254,7 +254,11 @@ nbitslen(::Type{T}, len, offset) where {T<:IEEEFloat} =
     min(cld(precision(T), 2), nbitslen(len, offset))
 # The +1 here is for safety, because the precision of the significand
 # is 1 bit higher than the number that are explicitly stored.
-nbitslen(len, offset) = len < 2 ? 0 : ceil(Int, log2(max(offset-1, len-offset))) + 1
+nbitslen(len, offset) = len < 2 ? 0 : ceil_log2(max(offset-1, len-offset)) + 1
+
+ceil_log2(n::Integer) = ceil(Int, log2(n))
+ceil_log2(n::Int64) = 64 - leading_zeros(n - Int64(1))
```

Comment:
`8*sizeof(T)`?

Reply:
I was trying to remember what that was called! Can write `Union{Int32, Int64}` but is it worth doing any further integers?

Reply:
IMO we should probably have this for `Base.BitSigned`